lib/src/realm_object.dart

```diff
@@ -297,6 +299,8 @@ extension RealmObjectInternal on RealmObject {
 
   RealmObjectHandle get handle => _handle!;
   RealmAccessor get accessor => _accessor;
+
+  String toJson() => realmCore.objectToString(this);
```

Comment:
moving this to test.dart since this is the only place using it. Since every tests imports test.dart this method will be available in the other test files as well. test.dart will need to import realm_core.dart

Reply:
I mean code in `lib` should really be code used in the SDK. If tests need to call core we should do it there in such cases. Its not ideal since until now it was not needed but we need it now for `toJson` and it seems for clear app cache as well